provider/hetzner.go

```diff
@@ -0,0 +1,206 @@
+/*
+	by Vladimir Smagin, 2020
+	vlad@blindage.org
+*/
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	hclouddns "git.blindage.org/21h/hcloud-dns"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/plan"
+)
+
+const (
+	hetznerCreate = "CREATE"
+	hetznerDelete = "DELETE"
+	hetznerUpdate = "UPDATE"
+	hetznerTTL    = 600
+)
+
+type HetznerChanges struct {
+	Action            string
+	ZoneID            string
+	ZoneName          string
+	ResourceRecordSet hclouddns.HCloudRecord
+}
+
+type HetznerProvider struct {
+	Client       hclouddns.HCloudDNS
+	domainFilter endpoint.DomainFilter
+	DryRun       bool
+}
+
+func NewHetznerProvider(ctx context.Context, domainFilter endpoint.DomainFilter, dryRun bool) (*HetznerProvider, error) {
+	token, ok := os.LookupEnv("HETZNER_TOKEN")
+	if !ok {
+		return nil, fmt.Errorf("No token found")
```

Comment:
@21h In this case I propose to use pure `errors` library instead of `fmt`

```go
import "errors"
```

```suggestion
		return nil, errors.New("No token found")
```

Reply:
More examples how can you use errors: https://github.com/kubernetes-sigs/external-dns/blob/master/provider/ovh/ovh.go#L40